src/wallet/wallet.cpp

```diff
@@ -100,8 +100,51 @@ libzcash::PaymentAddress CWallet::GenerateNewZKey()
     return addr;
 }
 
+// Generate a new Sapling spending key and return its public payment address
+SaplingPaymentAddress CWallet::GenerateNewSaplingZKey()
+{
+    AssertLockHeld(cs_wallet); // mapSaplingZKeyMetadata
+    
+    auto sk = SaplingSpendingKey::random();
+    auto fvk = sk.full_viewing_key();
+    auto addr = sk.default_address();
+    
+    // Check for collision, even though it is unlikely to ever occur
```

Comment:
Technically it is also possible for a collision to occur on ivk, even though fvk does not collide.